lib/core/fs/wrapper/source_file.dart

```diff
@@ -80,6 +80,17 @@ class SourceFile {
   String relativePath({String origin = '', bool filename = true}) =>
       p.joinAll([origin, ...path, filename ? '$name.$extension' : '']);
 
+  /// Creates a copy of this file enuring that the data is stored in memory.
+  Future<SourceFile> asMemoryData() async {
+    return SourceFile(
+      entry,
+      path,
+      name,
+      extension,
+      data: await data,
```

Comment:
When sending a source file through a send port, the data has to be in-memory. Theoretically a file reference would be fine too, but that doesn't work with the getter. After sending the file stored in cache (on disk). After that the garbage collector should remove it, because we do not store a reference to it afterwards.

A DoS attack based on this is very unlikely. Before someone could send enough files to the server to fill the memory the garbage collector would have already removed most of them because RAM is much faster than networking

Reply:
You are probably right, but why cant we send file handles? We would have to somehow clean up if the other side stoped working all of a sudden, thats right.

Reply:
Simplicity. When we want to use file handles, we'd need to create **another** abstraction around the same data as `SourceFile` and `FileUpdate` does.